### DISCUSSION TITLE
Why pulsar retention quota must exceed configured backlog quota

### DISCUSSION THREAD

**Discussion by mnit016** [apache/pulsar#17697] [Q&A] [answered]
Hello everyone, 

I'm new to Pulsar, recently I get into a problem when configure the backlog/retention quota for namespace.

The problem:
retention: {time: -1, size: 10M} => then I can set backlog quota as much as I want.
But if retention: {time: 1, size: 10M} => when I set backlog quota exceed 10M, it said "Backlog Quota exceeds configured Retention quota for namespace. Please increase retention quota and retry", I have no idea why.

Could someone please help to explain for me?
Thanks

**Comment by Jason918** [comment, marked as answer]
> retention: {time: -1, size: 10M} => then I can set backlog quota as much as I want.

This seems to be a bug. It's introduced in #10401 since v2.8.0

**Reply by michaeljmarshall** [reply]
@mnit016 - are you willing to create a GitHub issue for this bug? Thanks!

**Reply by michaeljmarshall** [reply]
I just discovered the "create issue from discussion" button. I created an issue for this discussion.

**Reply by Jason918** [reply]
I have pushed a fix https://github.com/apache/pulsar/pull/17706

**Reply by mnit016** [reply]
Hello @Jason918 , @michaeljmarshall , but I still confused, could you please explain for me why backlog quota couldn't exceed the retention? Is there any risk if we set backlog quota larger than retention?

**Reply by Jason918** [reply]
> Hello @Jason918 , @michaeljmarshall , but I still confused, could you please explain for me why backlog quota couldn't exceed the retention? Is there any risk if we set backlog quota larger than retention?

Messages will be deleted if they meet the constrains of retention policy. So if backlog quota is larger than retention, it just won't take effect.
